<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>882</width>
    <height>463</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Minecraft server scanner</string>
  </property>
  <property name="windowIcon">
   <iconset resource="Icon.qrc">
    <normaloff>:/images/icon.png</normaloff>:/images/icon.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">*{
	background:rgb(255,255,255);
}
#startButton
{
	background:rgb(0,133,252);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#startButton:pressed
{
	background:rgb(0,122,228);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#startButton:disabled
{
	background:rgb(143,188,228);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#pauseButton
{
	background:rgb(0,133,252);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#pauseButton:pressed
{
	background:rgb(0,122,228);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#pauseButton:disabled
{
	background:rgb(143,188,228);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#stopButton
{
	background:rgb(0,133,252);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#stopButton:pressed
{
	background:rgb(0,122,228);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
#stopButton:disabled
{
	background:rgb(143,188,228);
	border-radius:8px;
	font-size:15px;
	font-family:Century Gothic, sans-serif;
	color:rgb(255,255,255);
}
QGroupBox
{
	font-family:Century Gothic, sans-serif;
}
QMenu::item:selected
{
	background-color:rgb(204,232,255);
	color:rgb(0,0,0);
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QTableView" name="tableView"/>
    </item>
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>264</width>
          <height>143</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>264</width>
          <height>143</height>
         </size>
        </property>
        <property name="title">
         <string>Settings</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="startIpLabel">
               <property name="text">
                <string>Starting IP address:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="startIpLineEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="endIpLabel">
               <property name="text">
                <string>Ending IP address:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="endIpLineEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QLabel" name="portLabel">
               <property name="text">
                <string>Check specific port:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="portSpinBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>100000</number>
               </property>
               <property name="value">
                <number>25565</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="threadsLabel">
               <property name="text">
                <string>Number of Threads:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="threadsSpinBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>129</width>
                 <height>20</height>
                </size>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>200</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="startButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pauseButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Pause</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="stopButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="sizeIncrement">
           <size>
            <width>83</width>
            <height>30</height>
           </size>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>882</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="CustomMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave_results"/>
    <addaction name="actionOpen_file"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSave_results">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save results</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionOpen_file">
   <property name="text">
    <string>Open file</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomMenu</class>
   <extends>QMenu</extends>
   <header>custommenu.hpp</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="Icon.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>440</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
